File: src/apps/Andriod/app/src/main/java/com/dkellycollins/triggerfinger/data/daos/IEnemyDao.java
package com.dkellycollins.triggerfinger.data.daos;

import com.dkellycollins.triggerfinger.data.entity.IEnemy;

/**
 * Provides the ability to manage enemy data.
 */
public interface IEnemyDao extends IActivityDao {

    int create(int collidableId);

    Iterable<IEnemy> retrieve();
    IEnemy retrieve(int id);

    void update(int id, int collidableId);

    void delete(int id);

}

File: src/apps/Android-Kotlin/app/src/main/kotlin/com/dkellycollins/triggerfinger/data/daos/IEnemyDao.kt
package com.dkellycollins.triggerfinger.data.daos

import com.dkellycollins.triggerfinger.data.entity.IEnemy

/**
 * Provides the ability to manage enemy data.
 */
interface IEnemyDao : IActivityDao {

    fun create(collidableId: Int): Int

    fun retrieve(): Iterable<IEnemy>
    fun retrieve(id: Int): IEnemy

    fun update(id: Int, collidableId: Int)

    fun delete(id: Int)

}
